rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/widget/OneScanCovidRDTBarcodeFactoryTest.java

```diff
@@ -108,6 +114,40 @@ public void testOnActivityResultShouldPopulateCorrectData() throws Exception {
         // verify correct action on back-press
         oneScanCovidRDTBarcodeFactory.onActivityResult(JsonFormConstants.BARCODE_CONSTANTS.BARCODE_REQUEST_CODE, Activity.RESULT_CANCELED, intent);
         Mockito.verify(formFragment).setMoveBackOneStep(true);
+
+        Intent invalidIntent = new Intent();
+        invalidIntent.putExtra(Constants.Config.ENABLE_BATCH_SCAN, true);
+        invalidIntent.putExtra("data", "");
+        oneScanCovidRDTBarcodeFactory.onActivityResult(JsonFormConstants.BARCODE_CONSTANTS.BARCODE_REQUEST_CODE, Activity.RESULT_OK, invalidIntent);
+        oneScanCovidRDTBarcodeFactory.onActivityResult(Activity.RESULT_OK, JsonFormConstants.BARCODE_CONSTANTS.BARCODE_REQUEST_CODE, null);
+    }
+
+    @Test
+    public void testLaunchBarcodeScanner() throws Exception {
+        MaterialEditText materialEditText = Mockito.mock(MaterialEditText.class);
+        InputMethodManager inputManager = Mockito.mock(InputMethodManager.class);
+        Activity activity = Mockito.mock(Activity.class);
+        Mockito.when(activity.getSystemService(ArgumentMatchers.anyString())).thenReturn(inputManager);
+        Whitebox.invokeMethod(oneScanCovidRDTBarcodeFactory, "launchBarcodeScanner", activity, materialEditText, "");
+        Mockito.verify(activity).startActivityForResult(ArgumentMatchers.any(Intent.class), ArgumentMatchers.anyInt());
+    }
+
+    @Test
+    public void testGetBarcodeValsAsCSV() throws Exception {
+        Barcode barcode = new Barcode();
+        barcode.displayValue = "";
+        Intent intent = new Intent();
+        intent.putExtra(JsonFormConstants.BARCODE_CONSTANTS.BARCODE_KEY, barcode);
+        String displayValue = Whitebox.invokeMethod(oneScanCovidRDTBarcodeFactoryOriginal, "getBarcodeValsAsCSV", intent);
+        Assert.assertTrue(StringUtils.isEmpty(displayValue));
```

Comment:
Could you test this with an intent that has data as well?

Reply:
This would also remove the need for this https://github.com/onaio/rdt-standard/pull/656/files#diff-7c8f82faba856576126f062799a7e505ed0666e6f93cdd2541eec5e2733dd4f0R146.

Reply:
got fixed. i have used mocked barcode